### PR TITLE
Using store abstractions in tools + supporting segmented logs in tools

### PR DESCRIPTION
**Pull request by vgkholla**
ambry-tools had some repeated code for reading index segments. This commit makes use of the `IndexSegment` class to read the index file and thus unifies the code that enables interpretation of index segment formats

This commit also enables support for segmented logs in ambry tools and enables them to work with segmented or non segmented logs

This commit also removed the option to verify consistency of log entries with index. This functionality is (almost) fully covered in the code that verifies the consistency of index entries with the log and thus is no longer required.

This commit also removes the `IndexConsistencyChecker` tool which is no longer relevant and has been replaced by better tools (compareIndexEntriesWithLog).

**Primary reviewer : @nsivabalan**
Formatted and `./gradlew build` succeeded. 
Tested by verifying some segmented logs.

### CODE REVIEW

File: ambry-tools/src/main/java/com.github.ambry/store/DumpData.java [hunk 1]
```diff
@@ -585,91 +582,49 @@ private void logRangesNotCovered(Map<Long, Long> coveredRanges, long indexEndOff
...
+        if (isDeleted) {
+          long originalOffset = value.getOriginalMessageOffset();
+          if (originalOffset != -1) {
+            if (!coveredRanges.containsKey(originalOffset)) {
               if (startOffset.getOffset() > originalOffset) {
                 logger.trace("Put Record at {} with delete msg offset {} ignored because it is prior to startOffset {}",
-                    originalOffset, blobValue.getOffset(), startOffset);
+                    originalOffset, value.getOffset(), startOffset);
               } else {
                 randomAccessFile.seek(originalOffset);
```
Comment (nsivabalan) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpData.java]: not required.

File: ambry-store/src/main/java/com.github.ambry.store/IndexValue.java [hunk 2]
```diff
@@ -113,6 +113,12 @@ void setNewSize(long size) {
   ByteBuffer getBytes() {
     return value;
   }
+
+  @Override
+  public String toString() {
+    return "Offset: " + offset + ", Size: " + getSize() + ", Flags: " + getFlags() + ", ExpiresAtMs: "
```
Comment (vgkholla) [inline ambry-store/src/main/java/com.github.ambry.store/IndexValue.java]: Done.

File: ambry-tools/src/main/java/com.github.ambry/store/DumpData.java (hunk 1 above)
Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpData.java]: Will leave as is for now - can figure out what needs to be removed later.

Comment (nsivabalan) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpData.java]: You can remove this. I have verified that it is safe to remove.

Comment (vgkholla) [inline ambry-tools/src/main/java/com.github.ambry/store/DumpData.java]: Done

**Comment by nsivabalan** [comment]
LGTM. Merging.

**Pull request merged**